src/WingetCreateCLI/Program.cs

```diff
@@ -48,25 +60,46 @@ private static async Task<int> Main(string[] args)
                 typeof(CacheCommand),
                 typeof(ShowCommand),
             };
-            var parserResult = myParser.ParseArguments(args, types);
 
-            BaseCommand command = parserResult.MapResult(c => c as BaseCommand, err => null);
+            var baseCommandsParserResult = myParser.ParseArguments(args, types);
+            BaseCommand parsedCommand = baseCommandsParserResult.MapResult(c => c as BaseCommand, err => null);
 
-            if (command == null)
+            if (parsedCommand == null)
             {
-                DisplayHelp(parserResult as NotParsed<object>);
-                DisplayParsingErrors(parserResult as NotParsed<object>);
+                // In case of unsuccessful parsing, check if user tried to run a valid command.
+                bool isBaseCommand = baseCommandsParserResult.TypeInfo.Current.BaseType.Equals(typeof(BaseCommand));
+
+                /* Parse root options.
+                 * Adding '--help' is a workaround to force parser to display the options help text when no arguments are passed.
+                 * This makes rootOptionsParserResult to be a NotParsed object which makes HelpText.AutoBuild print the correct help text.
+                 * This is done since the parser does not print the correct help text on a successful parse.
+                  */
+                ParserResult<RootOptions> rootOptionsParserResult = myParser.ParseArguments<RootOptions>(

```

Comment:
The reason why this results in a NotParsed object even when I've specified a Help Option in RootOptions.cs is because it seems that the parser specifically reserves the `--help` and `--version` keywords. I've only listed Help Option in RootOptions.cs so that I can print a custom help text for it in the CLI output. I realize there is AutoHelp = true available in HelpText.AutoBuild method but that does not let me customize its helptext message.